pkg/distributor/distributor.go

```diff
@@ -355,20 +355,16 @@ func (d *Distributor) sendSamplesErr(ctx context.Context, ingester *ring.Ingeste
 	}
 	c := h.(ingester_client.IngesterClient)
 
-	req := &client.WriteRequest{
-		Timeseries: make([]client.TimeSeries, 0, len(samples)),
+	req := client.WriteRequest{
+		Timeseries: make([]client.PreallocTimeseries, len(samples)),
 	}
-	for _, s := range samples {
-		req.Timeseries = append(req.Timeseries, client.TimeSeries{
-			Labels:  s.labels,
-			Samples: []client.Sample{s.sample},
-		})
+	for i, s := range samples {
+		req.Timeseries[i].Labels = s.labels
+		req.Timeseries[i].Samples = s.samples
 	}
 
-	err = instrument.TimeRequestHistogram(ctx, "Distributor.sendSamples", sendDuration, func(ctx context.Context) error {
-		_, err := c.Push(ctx, req)
-		return err
-	})
+	_, err = c.Push(ctx, &req)
```

Comment:
No more instrumentation around the sendDuration?

Reply:
I didn't think it was useful and was obscuring my profiling, so I removed it.  I can add it back (probably as some gRPC client middleware) if you like?

Reply:
If you take this out, what stats are there to tell you the difference between the 2 results we accept and the 3rd that we discard?

Reply:
As Bryan mentioned, it can be useful for knowing if that third call is taking much longer than the other two, especially if it is timing out. `ingesterAppendFailures` is still there for that, or Jaeger traces, but the metric has been nice for me.